Sources/CvPlot.cpp

```diff
@@ -2636,58 +2446,51 @@ void CvPlot::updateSight(bool bIncrement, bool bUpdatePlotGroups)
 		/************************************************************************************************/
 	}
 
-	// Owned
 	if (isOwned())
 	{
 		changeAdjacentSight(getTeam(), GC.getDefineINT("PLOT_VISIBILITY_RANGE"), bIncrement, NULL, bUpdatePlotGroups);
 	}
 
-	pUnitNode = headUnitNode();
-
-	// Unit
-	while (pUnitNode != NULL)
+	foreach_(CvUnit* unit, units())
 	{
-		pLoopUnit = ::getUnit(pUnitNode->m_data);
-		pUnitNode = nextUnitNode(pUnitNode);
-		
-		if (pLoopUnit != NULL)
+		if (unit != NULL)
```

Comment:
Not needed.

Reply:
Odd debugging code there. 
You see the else? Odd I know.

	foreach_(CvUnit* unit, units())
	{
		if (unit != NULL)
		{
			if (bIncrement)
			{
				unit->changeDebugCount(1);
			}
			else
			{
				unit->changeDebugCount(-1);
			}
			
			changeAdjacentSight(unit->getTeam(), unit->visibilityRange(), bIncrement, unit, bUpdatePlotGroups);
		}
		else
		{
			verifyUnitValidPlot();
			break;
		}
	}

Reply:
Yeah no idea about this code, but unit should never be null in any context!

Reply:
I added similar FAsserts to CvSelectionGroup. Can I remove them?